tests/system/libraries/NotepadLib.py

```diff
@@ -29,40 +30,77 @@
 from robot.libraries.Process import Process as _ProcessLib
 from AssertsLib import AssertsLib as _AssertsLib
 import NvdaLib as _NvdaLib
+import WindowsLib as _WindowsLib
 
 builtIn: BuiltIn = BuiltIn()
 opSys: _OpSysLib = _getLib('OperatingSystem')
 process: _ProcessLib = _getLib('Process')
 assertsLib: _AssertsLib = _getLib('AssertsLib')
+windowsLib: _WindowsLib = _getLib('WindowsLib')
 
 
 # In Robot libraries, class name must match the name of the module. Use caps for both.
 class NotepadLib:
 	_testFileStagingPath = _tempfile.mkdtemp()
 	_testCaseTitle = "test"
 
-	def __init__(self):
-		self.notepadHandle: _Optional[int] = None
+	# Use class variables for state that should be tied to the RF library instance.
+	# These variables will be available in the teardown
+	notepadWindow: _Optional[Window] = None
+	processRFHandleForStart: _Optional[int] = None
 
 	@staticmethod
 	def _getTestCasePath(filename):
 		return _pJoin(NotepadLib._testFileStagingPath, filename)
 
 	def exit_notepad(self):
 		spy = _NvdaLib.getSpyLib()
-		spy.emulateKeyPress('alt+f4')
-		process.wait_for_process(self.notepadHandle, timeout="1 minute", on_timeout="continue")
+		builtIn.log(
+			# True is expected due to /wait argument.
+			"Is Start process still running (True expected): "
+			f"{process.is_process_running(NotepadLib.processRFHandleForStart)}"
+		)
+
+		if _getForegroundHwnd() == NotepadLib.notepadWindow.hwndVal:
+			builtIn.log("Test case in foreground, trying to close")
+			spy.emulateKeyPress('alt+f4')
+			process.wait_for_process(
+				NotepadLib.processRFHandleForStart,
+				timeout="1 minute",
```

Comment:
this is quite a long timeout for notepad to close

Reply:
Yep, I'll reduce this to 10 seconds.